src/stactools/core/utils/raster_footprint.py

```diff
@@ -0,0 +1,296 @@
+    asset_names: Optional[List[str]] = None,
+    scale: int = DEFAULT_SCALE,
+    precision: int = DEFAULT_PRECISION,
+    densification_factor: Optional[int] = None,
+    simplify_tolerance: Optional[float] = None,
+    no_data: Optional[int] = None,
+) -> bool:
+    """
+    Accepts an Item and an optional list of asset names within that item, and updates
+    the geometry of that Item in-place with the data footprint derived from the first
+    of the assets that exists in the Item.
+
+    Two important operations during this calculation are the densification of the
+    footprint in the native CRS and simplification of the footprint after reprojection.
+    If the initial low-vertex polygon in the native CRS is not densified, this can result
+    in a reprojected polygon that does not accurately represent the data footprint. For
+    example, a MODIS asset represented by a rectangular 5 point Polygon in a sinusoidal
+    projection will reproject to a parallelogram in EPSG 4326, when it would be more
+    accurately represented by a polygon with two parallel sides and two curved sides.
+    The difference between these representations is even greater the further away from
+    the meridian and equator the asset is located.
+
+    After reprojection to EPSG 4326, the footprint may have more points than desired.
+    This can be simplified to a polygon with fewer points that maintain a minimum distance
+    between each point.
+
+    Args:
+        item (Item): The PySTAC Item to extend.
+        asset_names (Optional[List[str]]): The names of the assets for which to extract footprints.
+        scale (int): A factor to scale out the shape to. Defaults to 1.
+        precision (int): The number of decimal places to include in the coordinates for the
+            reprojected geometry. Defaults to 3 decimal places.
+        densification_factor (Optional[int]): The factor by which to increase point density within
+            the polygon.
+            A factor of 2 would double the density of points (placing one new point between each
+            existing pair
+            of points), a factor of 3 would place two points between each point, etc.
+        simplify_tolerance (Optional[float]): All points in the simplified object will be within
+            the tolerance distance of the original geometry, in degrees.
+        no_data(Optional[int]): explicitly set the no data value if not in image metadata
+
+    Returns:
+        bool: True if the extent was successfully updated, False if not
+    """
+    asset_name_and_extent = next(
+        data_footprints_for_data_assets(
+            item,
+            asset_names=asset_names,
+            scale=scale,
+            precision=precision,
+            densification_factor=densification_factor,
+            simplify_tolerance=simplify_tolerance,
+            no_data=no_data,
+        ),
+        None,
+    )
+    if asset_name_and_extent is not None:
+        _, extent = asset_name_and_extent
```

Comment:
I'm not clear on this. What is the purpose of allowing multiple asset_names if only the first one (returned by `next`) is used?

Reply:
in case your items have different assets, so they're in decreasing preference, e.g., "blue_10m", "blue_60m", etc.

Reply:
So if the footprint fails for the first asset, it tries on the second asset, and so on?

Reply:
yes

Reply:
or more likely than failing is that it doesn't exist